packages/react-server/src/ReactFizzServer.js

```diff
@@ -199,9 +211,11 @@ export opaque type Request = {
   nextSegmentId: number,
   allPendingTasks: number, // when it reaches zero, we can close the connection.
   pendingRootTasks: number, // when this reaches zero, we've finished at least the root boundary.
+  resources: Resources,
   completedRootSegment: null | Segment, // Completed but not yet flushed root segments.
+  rootDidFlush: boolean,
```

Comment:
Why do we need this now? What can no longer be inferred?

Reply:
At the moment `flushCompletedQueues` can be called before `performWork`. we could let `flushImmediateResources` get called b/c it should be a noop given there are no resources until we've done work but it feels technically incorrect. In fact I'd say it's incorrect to check all the boundaries for flushing as well given they should never flush until the root is ready.

we could use the presence of `completedRootSegment` as the signifier of whether we can even do a flush but this is set back to null after it flushes so we'd either need to get cute with distinguishing between undefined and null or have rootDidFlush or a similar flag like rootDidComplete.

Reply:
removed